_summaries/sukhbaatar2015end-to-end_1.md

```diff
@@ -1,19 +1,76 @@
 ---
 layout: summary
 title: Summary
-paper: {{paper_tag}}
+paper: sukhbaatar2015end-to-end
 # Please fill out info below
-author: # Your GitHub id
-score: # How did you like this paper 0(dislike) to 10(love)
+author: DartingMelody
+score: 7
 ---
 
-TODO: Summarize the paper:
-* What is the core idea?
-* How is it realized (technically)?
-* How well does the paper perform?
-* What interesting variants are explored?
+**What is the core idea?**
+
+The paper introduces a novel recurrent neural network with an explicit large memory and a recurrent attention mechanism. The model reads recurrently from memory multiple times before giving the results. This model is trained end-to-end and therefore requires comparitively less supervision during training. Hence, it can also be applied to realistic settings. The paper also shows experimentally that multiple hops over the long-term memory leads to good performance of the model. 
+
+**How is it realized (technically)?**
+
+The Model takes as input $$x_{1},...,x_{i}$$ (to store in memory), query _q_ and outputs answer _a_. The model writes all _x_ to the memory up to a fixed buffer size and creates continuous representation of _x_ and _q_. 
+
+*Single layer:*
+
+1) The input set{$$x_{i}$$} is embedded with matrix _A_ ( _d_ x _V_) to give memory vectors {$$m_{i}$$}. 
+2) The query _q_ is also embedded with matrix _B_ (_d_ x _V_) to give an internal state u.
+3) On computing match between each memory {$$m_{i}$$} and _u_ by taking softmax of their inner product which results into the probability vector p over the inputs i.e. $$p_{i} = Softmax(u^{T}m_{i})$$
+4) Each $$x_{i}$$ has a corresponding mapping to an ouput vector $$c_{i}$$ (using another embedding matrix _C_).
+5) The final output is the sum of the weighted probabilities {$$p_{i}$$}  over the transformed input $$c_{i}$$ i.e. $$o = \sum_{i}^{}p_{i}c_{i}$$
+6) The output vector _o_ and embedding vector _u_ is added and multiplied by the final weight matrix _W_ which is then passed on to the softmax to predict the output i.e. $$\hat{a} = Softmax(W(o + u))$$
+7) As the input to output function is smooth, the gradient an be easily computed of this continuous function. All three embedding matrices _A_, _B_, _C_ and weight matrix _W_ are learnt by minimizing cross entrophy loss. 
+
+*Multiple layer:*
+
+1) For layers above the first layer, the input is the sum of the output $$o^{k}$$ and the input $$u^{k}$$ i.e. $$u^{k+1} =u^{k}+o^{k}$$
+2) Each layer has it's own embedding matrices _A_ and _C_ but have constraints to ease training and reduce hyperparameters. 
+3) At final layer, the output is $$\hat{a} = Softmax(W u^{K+1}) = Softmax(W (o^{K} + u^{K} ))$$
```

Comment:
Just making note of this again, but instead of the final layer maybe use "final hop" because nothing is really different from the first iteration and the last iteration-- just the u^k and o^k are now representative of some sequence.